src/Tasks/Microsoft.NET.Build.Tasks/GenerateDepsFile.cs

```diff
@@ -103,6 +113,11 @@ protected override void ExecuteCore()
                 .WithReferenceAssembliesPath(FrameworkReferenceResolver.GetDefaultReferenceAssembliesPath())
                 .Build();
 
+            if (compileFilesToSkip.Any() || runtimeFilesToSkip.Any())
+            {
+                dependencyContext = TrimFilesToSkip(dependencyContext);
```

Comment:
Sure, it would be possible.  I think it's probably not worth it as there's risk of changing the behavior while refactoring the code, plus the naming is potentially confusing (should it be `WithFilesToSkip` or `WithoutFilesToSkip`?).